src/reanimated2/core.ts

```diff
@@ -213,6 +213,22 @@ if (!isWeb() && isConfigured()) {
       info: runOnJS(capturableConsole.info),
     };
     _setGlobalConsole(console);
+
+    const nativeRequestAnimationFrame = global.requestAnimationFrame;
+    let callbacks: Array<(time: number) => void> = [];
+    global.requestAnimationFrame = (
```

Comment:
I don't like the idea that here we override the JSI binding with a regular JS function. I would like to suggest two improvements:
1. Rename the JSI binding to `nativeRequestAnimationFrame` to resolve the name conflict and discourage future developers from using it. We already provide the JS function `requestAnimationFrame` which satisfies the semantics (i.e. can be called multiple number of times) so there's no reason to expose an identical API (except for batching) from the native side. We can treat `nativeRequestAnimationFrame` as Reanimated-specific implementation detail.
2. (Optionally) Add a check on native side (in debug mode only) that `nativeRequestAnimationFrame` is called no more than once per single frame to prevent using it directly in the future.